src/System.Slices/System/Buffers/Memory.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Represents a Span<byte> factory</byte>
+    /// </summary>
+    /// <remarks>
+    /// This struct is not safe for multithreaded access, i.e. it's subject to struct tearing that can result in unsafe memory access. 
```

Comment:
Cheap because I was replacing code that was slicing up spans before so this ended up being the alternative. I need to allocate say a 4K chunk, have pieces of code write into it then get the underlying pointer (lets call that unwrapping) so I can pass it to a native API like libuv for outgoing writes. Maybe the pattern should be to expose a `Span<byte>` for reading an writing but still provide access to `Memory<byte>` just for unwrapping. 

I'd be OK making it a class if there was only one of them and they were pooled.